### PR TITLE
DI (IOC) setup, support decorators, use single model for db and api toward UI.

### PR DESCRIPTION
**Pull request by nakulgopal1013**
# Changelog

- DI (IOC) setup
- support decorators
- use single model for db and api toward UI.

## What can be improved

- How DI container is setup ( check it's import)
- Render lists using FlatList
- Possibly remove DTO from Axios to API layer.
-  Solve Cyclic dep in Database and Schema class.

NOTE : This PR is based on #1

### CODE REVIEW

File: src/core/network/axios/axios_operations.ts [hunk 1]
```diff
@@ -11,8 +16,12 @@ class AxiosOperations implements IAxiosOperations {
     requestFunction: () => Promise<AxiosResponse<T>>,
   ): Promise<AxiosResponse<T>> {
     try {
+      if (store.getState().internet.isConnected) {
```
Comment (AkritiSaraogi10) [inline src/core/network/axios/axios_operations.ts:19]: We added the if-else condition for calling remote or Realm APIs, but its not required now, we can remove it.

Comment (nakulgopal1013) [reply src/core/network/axios/axios_operations.ts]: Removed.

File: src/shared/presentation/components/searchbar.tsx
[the remark this answers is not in the document]
Comment (nakulgopal1013) [reply src/shared/presentation/components/searchbar.tsx:1]: Deleted

File: src/features/posts/presentation/screens/post_screen_manager.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  };
+
+  const handleEditSave = (id: string) => {
+    const postChanged = editedPosts.filter(p => p._id.toString() === id)[0];
+    postUseCase.updatePost(postChanged as PostSchema);
+  };
+
+  const handleDeletePost = (id: string) => postUseCase.deletePost(id ?? '');
+
+  const handleInputChange2 = (key: number, value: string) => {
```
[the remark this answers is not in the document]
Comment (nakulgopal1013) [reply src/features/posts/presentation/screens/post_screen_manager.tsx:89]: done

**Pull request merged**